docs/src/examples/redis-clustering-aware.md

```diff
@@ -0,0 +1,156 @@
+# Redis Clustering with cluster aware client
+
+The following guide shows you how to configure Shotover to support proxying Redis cluster *aware* clients to [Redis cluster](https://redis.io/topics/cluster-spec).
+
+## Overview
+
+In this example, we will be connecting to a Redis cluster that has the following topology:
+
+* `127.0.0.1:7000`
+* `127.0.0.1:7001`
+* `127.0.0.1:7002`
+* `127.0.0.1:7003`
+* `127.0.0.1:7004`
+* `127.0.0.1:7005`
+
+Shotover will listen on the loopback adapter and act as a sidecar for our application that speaks Redis. In this example we will use `redis-benchmark` with cluster mode enabled as our Redis cluster aware client application. Use the following [docker-compose.yml](https://github.com/shotover/shotover-proxy/blob/main/shotover-proxy/examples/redis-cluster-ports-rewrite/docker-compose.yml) to run the Redis cluster.
+
+```bash
+curl -L https://raw.githubusercontent.com/shotover/shotover-proxy/main/shotover-proxy/examples/redis-cluster-ports-rewrite/docker-compose.yml --output docker-compose.yml
+docker-compose -f docker-compose.yml up
+```
+
+## Configuration
+
+First we will modify our `topology.yaml` file to have a single Redis source. This will:
+
+* Define how Shotover listens for incoming connection from our client application (`redis-benchmark`).
+* Configure Shotover to connect to the Redis cluster via our defined contact points.
+* Configure Shotover to rewrite all Redis ports with our Shotover port when the cluster aware driver is talking to the cluster, through Shotover.
+* Connect our Redis Source to our Redis cluster sink (transform).
+
+```yaml
+{{#include ../../../shotover-proxy/examples/redis-cluster-ports-rewrite/topology.yaml}}
+```
+
+Modify an existing `topology.yaml` or create a new one and place the above example as the files contents.
+
+## Starting
+
+We can now start `shotover-proxy`, by running the following:
+
+```console
+./shotover-proxy --topology topology.yaml 
+```
+
+The `RedisSinkCluster` transform needs to be able to connect to the Redis cluster when it starts up. If it cannot, Shotover will exit with an error, indicating it couldn't connect to the contact points. 
+  
+## Testing
+
+With Shotover now up and running, we can test out our client application. Let's start it up!
+
+```console
+redis-benchmark -t set,get --cluster
```

Comment:
I dont think this will work as we havent specified port 6380.
As mentioned elsewhere we should probably just change shotover to use 6379 though.